### ISSUE TITLE
core.autocrlf is ignored when checking out using GitHub Desktop

### ISSUE THREAD

**Issue by saschanaz** [desktop/desktop#10666]
### Describe the bug

Cloning a new repo or switching branch using GitHub Desktop causes files to use CRLF even when `git config core.autocrlf` is `input`.

### Version & OS

2.5.5 on Windows 10 2004

### Steps to reproduce the behavior

1. Set `git config core.autocrlf` to `input`.
2. Clone https://github.com/foolip/mdn-bcd-collector/ via GitHub Desktop
3. Check line endings
4. `git switch modernizr && git switch main` via CLI
5. Check line endings again

### Expected behavior

Line endings should always be LF.

### Actual behavior

They become CRLF when checking out using GitHub Desktop.

**Comment by niik** [comment]
Hi @saschanaz, thanks for opening this.

GitHub Desktop currently ships with its own system-level Git config on Windows. See https://github.com/desktop/dugite-native/pull/313 for the reason why we're not inheriting the config. We'd like to start inheriting the configuration but we'll need a bit of work in dugite-native before we can do so.

For now the easiest workaround would be to set the `code.autocrlf` option on the global level instead, i.e. `git config --global core.autocrlf input`.

I hope that workaround will unblock you until we're able to make the necessary changes in dugite-native.

**Comment by tfabris** [comment]
@niik - This still occurs, and it also occurs on Macintosh. Your suggested workaround of:

```git config --global core.autocrlf input```

... does not work for this. I have done that, and yet GitHub Desktop is still pulling down the files in the wrong format. In the case of the Mac, it's taking files which are stored with Windows line endings  and changing them to Unix line endings after the files are pulled down.

Is there another workaround other than that one?

**Comment by sraillard** [comment]
The file path in Windows for the git configuration file is:
`C:\Users\USER_NAME\.gitconfig`

If you have installed git from cygwin, you'll have another git configuration file here (it will be ignored by Github Desktop):
`C:\cygwin64\home\USER_NAME\.gitconfig`

In the Github Desktop settings, it would be nice to have a shortcut to the git config file used.

**Comment by ivoros** [comment]
The last comment was made 2 years ago. Github Desktop still doesn't respect the existing line endings.
Is there any news on this issue?

**Comment by timsu92** [comment]
I have GitHub Desktop installed only and I don't have Git installed on Windows. Do I have to install Git to modify this settings?

I've tried to change settings in `C:\Users\USER_NAME\.gitconfig` with no luck.
```gitconfig
[core]
	autocrlf = false
```
Is there another way to fix this?

**Comment by DanielYamini** [comment]
Unsubscribe

**Comment by theodorejb** [comment]
I'm also having this issue. I couldn't figure out why tests were failing after cloning a repository, until I found GitHub Desktop didn't respect my global `core.autocrlf = input` setting and changed all the line endings to be Windows-specific.

**Comment by wojiushixiaobai** [comment]
- %localappdata%\GitHubDesktop\app-{your_github_desktop_version}\resources\app\git\etc\gitconfig

```
[core]
symlinks = false
autocrlf = false  # <-- fix this
fscache = true
```

Re checkout the code.

**Comment by saenic** [comment]
This workaround is good. But why does Github Desktop still use its own gitconfig file overwriting the existing global .gitconfig?
